### ISSUE TITLE
Navigator resize since upgrade 2.4 to 4.1

### ISSUE THREAD

**Issue by lucasnngit** [openseadragon/openseadragon#2373]
Hello, I recently upgrade my OSD version from 2.4 to 4.1 and before, when I resized the container div of the navigator, I always saw the image as a whole, without it being cropped. When I update to 4.1, I cannot make it works as the same. When I resize the navigator, the image does not update well and bounds are cropped.

Here is my OSD options for my viewer :
		springStiffness: 15,
		defaultZoomLevel: 1,
		preload: true,
		minZoomImageRatio: 1,
		maxZoomPixelRatio: 2,
		zoomPerClick: 1,
		zoomPerScroll: 2,
		showNavigationControl: false,
		preserveImageSizeOnResize: true,
		maxImageCacheCount: 600,
		showNavigator: true,
		navigatorBackground: "rgba(200, 200, 200, 0.7)",
		navigatorOpacity: 1,
		navigatorBorderColor: "none"
		
Have you an idea why ? Is it a bug in the new version ?

**Comment by pearcetm** [comment]
The behavior on resize has changed a couple of times since v2.4. Try `preserveImageSizeOnResize: false` - does that do what you want it to?

**Comment by iangilman** [comment]
This may or may not be related to https://github.com/openseadragon/openseadragon/pull/2347; probably not caused by it, but maybe another loose end that's similar. 

I tried reproducing this with just the settings given above, and it's working fine for me. How are you resizing the navigator? Can you share a repro?

**Comment by lucasnngit** [comment]
> The behavior on resize has changed a couple of times since v2.4. Try `preserveImageSizeOnResize: false` - does that do what you want it to?

I try it but nothing better, when I resize to have big width and a small height, it cropped the top and bottom, and vice versa.


> How are you resizing the navigator? Can you share a repro?

I only use the jquery resizable function on the container div like this :
$("#navigatorWrapper" + viewerId).resizable();

**Comment by lucasnngit** [comment]
I just before wrap the navigator in an other div like this :
 $("#" + viewer.osdViewer.navigator.id)
			.detach()
			.appendTo("#navigatorWrapper" + viewerId); 
			
And I checked if auto resize was correctly setted to true in the library

**Comment by pearcetm** [comment]
I just tried this with the code you provided (moving the navigator to a different container, making that container resizable), and it's working as expected. Can you put together an example that reproduces it in a codepen or similar?

**Comment by lucasnngit** [comment]
Ok, I found that if I did not set a size to the navigator, the navigator doesn't display well.

If I set a size like this :
viewer.navigator.setWidth(200);
viewer.navigator.setHeight(200);
It display well but when I resize the container div, the whole image is displayed in the navigator but the image is distorted.

Then when I set a width/height every "resize" detected, it works correctly :
$("#navigatorWrapper" + viewerId).resize(() => {
			const navWidth = $("#navigatorWrapper" + viewerId).width();
			const navHeight = $("#navigatorWrapper" + viewerId).height();
			viewer.navigator.setWidth(navWidth);
			viewer.navigator.setHeight(navHeight);
		});
	};

My project is a quit big so maybe it's a side effect which causes a different behavior from yours. But like that it works fine.


Thanks you very much for your support and your responsiveness.
